parachain/pallets/ethereum-beacon-client/src/lib.rs

```diff
@@ -388,6 +393,22 @@ pub mod pallet {
 		}
 
 		fn process_finalized_header(update: FinalizedHeaderUpdateOf<T>) -> DispatchResult {
+			let current_block_number = <frame_system::Pallet<T>>::block_number();
+			let last_finalized_block_number = <LatestFinalizedHeaderImportBlockNumber<T>>::get();
+			let weak_subjectivity_period_check =
+				last_finalized_block_number + T::WeakSubjectivityPeriod::get().into();
+
+			log::info!(
+				target: "ethereum-beacon-client",
+				"💫 Checking weak subjectivity period. Current block number :{:?} Weak subjectvitity period check: {:?}.",
+				current_block_number,
+				weak_subjectivity_period_check
+			);
+
+			if current_block_number > weak_subjectivity_period_check.into() {
+				panic!("weak subjectvity period expired, reinitialize bridge")
```

Comment:
Panicking is a big no-no. It will essentially brick the parachain and require a runtime upgrade to fix. Remember that we are going to be deploying our pallets onto the common-good BridgeHub parachain, which is shared with other bridges. We don't want to affect our neighbours.

Reply:
Whoops! 😄 I'll need to rethink this then. Perhaps set a flag in storage that is checked, so no further processing happens in the light client until this is resolved. 

I am unsure of how the light client would get the go-ahead from governance to sync from an accepted checkpoint? Should there be an extrinsic with a specific origin? Or how could that work?

Reply:
yeah a pallet dispatchable that can only be called by governance.

An example here: https://github.com/Snowfork/snowbridge/blob/317917cca47d62740ede01ceebc5d9e7d32ce24b/parachain/pallets/incentivized-channel/src/inbound/mod.rs#L164

Reply:
yeah, then maybe we just refactor the current dispatchable `sync_committee_period_update` a little bit for some [origin check](https://github.com/Snowfork/snowbridge/blob/317917cca47d62740ede01ceebc5d9e7d32ce24b/parachain/pallets/ethereum-beacon-client/src/lib.rs#L220) which we ignore now.

@vgeddes @claravanstaden and could we also allow to config these [beacon envs](https://github.com/Snowfork/snowbridge/blob/main/parachain/pallets/ethereum-beacon-client/src/config.rs#L13-L20) without hardcode?

Reply:
I think the complication here is that if we process a `sync_committee_period_update` as sent by governance, the last synced committee will be, for example, for period `50`. Say the light client went offline at period `40`. So now the relayers won't know to sync the committee periods `41` - `49`.

Besides syncing the sync committees, syncing the execution headers in the offline period should ideally be backwards as well, syncing from the provided governance checkpoint backwards to the last imported header. If we don't sync backwards and sync forwards (like the current relayer does), we could accidentally sync the forked chain and then never reach the provided checkpoint.

Reply:
> @vgeddes @claravanstaden and could we also allow to config these [beacon envs](https://github.com/Snowfork/snowbridge/blob/main/parachain/pallets/ethereum-beacon-client/src/config.rs#L13-L20) without hardcode?

Sure, in another PR though. 😄